test/insert.js

```diff
@@ -0,0 +1,49 @@
+var Pipeline = require( '..' )
+var assert = require( 'assert' )
+var fs = require( 'fs' )
+var path = require( 'path' )
+var Stream = require( 'stream' )
+
+describe( 'Pipeline.insert()', function() {
+
+  it( 'can insert a stream at specific indices', function( done ) {
```

Comment:
I added a transform that lowercases things – do you think that would suffice, or would you rather have something more complex?